source/server/lsp-servers.adb

```diff
@@ -1130,14 +1159,24 @@ package body LSP.Servers is
            LSP.Messages.Server_Notifications.Server_Notification'Class
          then
             --  This is a notification
-
-            Server_Backend.Before_Work (Message.all);
-            LSP.Messages.Server_Notifications.Server_Notification'Class
-              (Message.all).Visit (Notif_Handler);
-            Server_Backend.After_Work (Message.all);
+            begin
+               Server_Backend.Before_Work (Message.all);
+               LSP.Messages.Server_Notifications.Server_Notification'Class
+                 (Message.all).Visit (Notif_Handler);
+               Server_Backend.After_Work (Message.all);
+            exception
+               when E : others =>
```

Comment:
Why not to use Ada here?

when Property_Error | Precondition_Failure =>
  do something
  
when others =>
 do another

I found it more understandable then use of some magic subprogram to do something even more magic with exception identifiers.

Reply:
I don't think there is anything "magic" in the exception identifiers, this comes from http://www.ada-auth.org/standards/2xaarm/html/AA-11-4-1.html, paragraphs 7, 9, and 11 ;-) ;-)

The reason I'm using a separate function is that I would like to factorize the code between the two cases.